app/scripts/controllers/overview.js

```diff
@@ -188,12 +193,13 @@ angular.module('openshiftConsole')
     };
 
     var groupPods = function() {
-      if (!pods || !deployments) {
+      if (!pods || !deployments || !replicaSets) {
         return;
       }
 
-      $scope.podsByDeployment = PodsService.groupByReplicationController(pods, deployments);
-      $scope.monopodsByService = PodsService.groupByService($scope.podsByDeployment[''], services, showMonopod);
+      var allOwners = _.toArray(deployments).concat(_.toArray(replicaSets));
```

Comment:
agree the code is cleaner as is, and not worried about time complexity as much as the extra memory allocation of concating the arrays vs an array of arrays, even then its probably not enough to even fret over, especially if we fix the watch debounce